README.md

```diff
@@ -78,17 +85,45 @@ file { '/etc/someservice/db.yaml':
 
 This module provides the `conjur::secret` function, described above, and the `conjur` class, which can be configured to establish Conjur host identity on the node running Puppet.
 
-### Pre-established host identity
+### Conjur host identity with Host Factory
 
-For best security properties, use [conjurize](https://www.conjur.org/get-started/machine-identity.html) or a similar method to establish host identity before running Puppet to configure. This way Puppet master only ever handles a temporary access token instead of real, permanent Conjur credentials of the hosts it manages.
+If pre-establishing host identity is unfeasible, we instead recommend bootstrapping Conjur host identity using a [Host Factory](https://developer.conjur.net/reference/services/host_factory) token. Nodes inherit the permissions of the layer for which the Host Factory token was generated.
 
-If a host is so pre-configured, the settings and credentials are automatically obtained and used. In this case, all that is needed to use `conjur::secret` is a simple
+Note when used in this manner, the host factory token will only be used on the initial Puppet run, to establish identity which is then stored on the host. Subsequent runs will use that for Conjur authentication on the node side (at the time of collecting facts) and only provide the Puppet master with a temporary token to fetch the secrets with.
+
+To use a Host Factory token with this module, set variables `authn_login` and `host_factory_token`. Do not set the variable `authn_api_key` when using `host_factory_token`; it is not required. `authn_login` should have a `host/` prefix; the part after the slash will be used as the node’s name in Conjur.
 
 ```puppet
-include conjur
+class { 'conjur':
+  account            => 'mycompany',
+  appliance_url      => 'https://conjur.mycompany.com/',
+  authn_login        => 'host/redis001',
+  host_factory_token => Sensitive('3zt94bb200p69nanj64v9sdn1e15rjqqt12kf68x1d6gb7z33vfskx'),
+  ssl_certificate    => file('/etc/conjur.pem')
+}
+```
+
+By default, all nodes using this Puppet module to bootstrap identity with host_factory_token will have the following annotation set:
+
+```yaml
+puppet: true
 ```
 
-#### <a name="windows"></a>Pre-establish Host Identity on Windows Hosts
+### Conjur host identity with API key
+
+For one-off hosts or test environments it may be preferable to create a host in Conjur and then directly assign
+its Conjur identity in this module.
+
+```puppet
+class { 'conjur':
+  appliance_url   => 'https://conjur.mycompany.com/',
+  authn_login     => 'host/redis001',
+  authn_api_key   => Sensitive('f9yykd2r0dajz398rh32xz2fxp1tws1qq2baw4112n4am9x3ncqbk3'),
+  ssl_certificate => file('/conjur-ca.pem')
+}
+```
+
+#### <a name="windows"></a>Using Conjur host identity with API key on Windows hosts
```

Comment:
Is this link (`name=windows`) used anywhere? If not, we should remove it

Reply:
it's not used anywhere in this file, but I left it because I'm not sure where it ever _was_ referenced from (the docs? 🤷‍♀️). I agree I'd rather remove it, so maybe I will and worst case someone will end up in this doc and have to CTRL+F "windows"